File: apps/mybookkeeper/backend/app/schemas/transactions/reservation.py
import uuid
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel


class OccupancyResponse(BaseModel):
    total_nights: int
    reservation_count: int
    total_days: int
    occupancy_rate: Decimal


class ReservationRead(BaseModel):
    id: uuid.UUID
    organization_id: uuid.UUID
    property_id: uuid.UUID | None = None
    transaction_id: uuid.UUID | None = None

    res_code: str
    platform: str | None = None

    check_in: date
    check_out: date
    nights: int | None = None

    gross_booking: Decimal | None = None
    net_booking_revenue: Decimal | None = None
    commission: Decimal | None = None
    cleaning_fee: Decimal | None = None
    insurance_fee: Decimal | None = None
    net_client_earnings: Decimal | None = None
    funds_due_to_client: Decimal | None = None

    guest_name: str | None = None

    statement_period_start: date | None = None
    statement_period_end: date | None = None

    created_at: datetime

    model_config = {"from_attributes": True}

File: apps/mybookkeeper/frontend/src/shared/types/reservation/reservation.ts
export interface Reservation {
  id: string;
  organization_id: string;
  property_id: string | null;
  transaction_id: string | null;
  res_code: string;
  platform: string | null;
  check_in: string;
  check_out: string;
  nights: number;
  gross_booking: string | null;
  net_booking_revenue: string | null;
  commission: string | null;
  cleaning_fee: string | null;
  insurance_fee: string | null;
  net_client_earnings: string | null;
  funds_due_to_client: string | null;
  guest_name: string | null;
  statement_period_start: string | null;
  statement_period_end: string | null;
  created_at: string;
}
